include/highfive/H5DataType.hpp

```diff
@@ -136,6 +136,37 @@ class CompoundType : public DataType {
     void create(size_t size = 0);
 };
 
+template <typename T>
```

Comment:
in this case `T` should be the `AtomicType<T>` which you can use both for the underlying type and the value type. Or maybe since you need to pass to the constructor the underlying type as an instantiated type, you can remove the template parameter here. Either way, the type T and the template parameter of the underlying type should be the same. This should be restricted.

Reply:
Fixed